app/views/searches/index.html.erb

```diff
@@ -11,49 +11,98 @@
 </div>
 
 <script type="text/javascript">
-  $(function(){
-    $('.js-venue-list').imagepicker({
-      show_label: true,
-      changed: function () {
-          $('#filter-form').submit()
-      }
+$(function(){
+  $('.js-venue-list').imagepicker({
+    show_label: true,
+    changed: function () {
+      $('#filter-form').submit()
+    }
+  });
+
+  var difference = function(first, second) {
+    return $(first).not(second).get();
+  }
+
+  var updateSectionVisibility = function(oldTypes, newTypes) {
+    $.each(difference(oldTypes, newTypes), function (_, recordType) {
+      $('.' + recordType).removeClass("is-active")
     });
 
-    var difference = function(first, second) {
-       return $(first).not(second).get();
-    }
+    $.each(difference(newTypes, oldTypes), function (_, recordType) {
+      $('.' + recordType).addClass("is-active")
+    });
+  };
+
+  $('.js-type-list').imagepicker({
+    show_label: false,
+    changed: updateSectionVisibility
+  });
 
-    var updateSectionVisibility = function(oldTypes, newTypes) {
-      $.each(difference(oldTypes, newTypes), function (_, recordType) {
-        $('.' + recordType).removeClass("is-active")
-      });
+  $('.js-year-range').ionRangeSlider({
+    type: "double",
+    hide_min_max: true,
+    step: 1,
+    onFinish: function () {
+      $('#filter-form').submit()
+    }
+  });
 
-      $.each(difference(newTypes, oldTypes), function (_, recordType) {
-        $('.' + recordType).addClass("is-active")
-      });
-    };
+  // Init the image gallery
+  var gallery = $('#grid-gallery')
+  if (gallery.length) {
+    new CBPGridGallery(gallery.get(0));
+  }
 
-    $('.js-type-list').imagepicker({
-      show_label: false,
-      changed: updateSectionVisibility
+  $('.js-load-more-audios').click(function() {
+    var page = $(this).data("page")
+    $.get( "/audios?page=" + page, function( data ) {
+      if ( $(data).data("more") == true ) {
+        $('.js-load-more-audios').data("page", $(data).data("page"))
+      } else {
+        $('.audios .paging').hide();
+      }
+      $( ".audios-grid ul" ).append($(data).html());
     });
+  })
 
-    $('.js-year-range').ionRangeSlider({
-      type: "double",
-      hide_min_max: true,
-      step: 1,
-      onFinish: function () {
-        $('#filter-form').submit()
+  $('.js-load-more-videos').click(function() {
+    var page = $(this).data("page")
+    $.get( "/videos?page=" + page, function( data ) {
+      if ( $(data).data("more") == true ) {
+        $('.js-load-more-videos').data("page", $(data).data("page"))
+      } else {
+        $('.videos .paging').hide();
       }
+      $( ".videos-grid" ).append($(data).html());
+      VideoGrid.init();
```

Comment:
Yes. After adding new elements to the UL the JS needs to be re-initialize the grid or the videos don't play.